myst_parser/parsers/docutils_.py

```diff
@@ -268,6 +269,10 @@ def parse(self, inputstring: str, document: nodes.document) -> None:
         HTMLTranslator.depart_rubric = depart_rubric_html
         HTMLTranslator.visit_container = visit_container_html
         HTMLTranslator.depart_container = depart_container_html
+        HTMLTranslator.special_chars_no_amp = {  # needed by encode_fixed
+            k: v for k, v in HTMLTranslator.special_characters.items() if k != ord("&")
+        }
+        HTMLTranslator.encode = encode_fixed
```

Comment:
I'm afraid this is not something that cannot be merged; it would break restructuredtext parsing.

Monkey patching is done strictly as a last resort, and only for where the patch does not affect upstream use